hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/NetworkTransactionGetReceiptHandler.java

```diff
@@ -55,16 +69,82 @@ public Response createEmptyResponse(@NonNull final ResponseHeader header) {
         return Response.newBuilder().transactionGetReceipt(response).build();
     }
 
+    @Override
+    public boolean requiresNodePayment(@NonNull ResponseType responseType) {
+        return responseType == ANSWER_ONLY || responseType == ANSWER_STATE_PROOF;
+    }
+
+    @Override
+    public boolean needsAnswerOnlyCost(@NonNull ResponseType responseType) {
+        return COST_ANSWER == responseType;
+    }
+
     @Override
     public void validate(@NonNull final QueryContext context) throws PreCheckException {
         requireNonNull(context);
-        throw new UnsupportedOperationException("Not implemented");
+        final var query = context.query();
+        final var recordCache = context.createStore(RecordCache.class);
+        final TransactionGetReceiptQuery transactionGetReceiptQuery = query.transactionGetReceiptOrThrow();
+        if (transactionGetReceiptQuery.hasTransactionID()) {
+            final var receipts =
+                    recordCache.getReceipt(transactionGetReceiptQuery.transactionIDOrElse(TransactionID.DEFAULT));
```

Comment:
This is the right place to use `transactionIDOrThrow` because it is wrapped by a `hasTransactionID`. There is no chance that it will throw, or that it will use the "orElse" default.

Reply:
fixed it